website/release-notes-parser.ts

```diff
@@ -0,0 +1,53 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import fs from 'node:fs';
+
+export function parseNotes(
+  filename: string,
+  releaseVersion: string,
+): {
+  image: string;
+  blog: string;
+  title: string;
+  summary: string;
+} {
+  const fileContent = fs.readFileSync(filename, { encoding: 'utf-8' });
+  const resultText = fileContent.split('---', 3);
+
+  // get release image url
+  const imagePath = /image: (.+)\n/.exec(resultText[1]);
+  const imageName = imagePath ? imagePath[1] : '';
+  const imageUrl = imageName ? `https://podman-desktop.io${imageName}` : '';
+  const blogUrl = `https://podman-desktop.io/blog/podman-desktop-release-${releaseVersion}`;
+
+  // get summary part of release notes
+  const text = resultText[2]
+    .split('<!--truncate-->', 1)[0]
+    .replace("import ReactPlayer from 'react-player'\n", '') //remove react import statement
+    .replace(/!\[.+\)\n/, '') // remove image link
+    .replace(/\[Click here to download it\]\(.+\)!/, '') //remove download new version
+    .replace(/\n(\n)+/g, '\n')
+    .split('\n'); // change all multi-newlines chars to one newline char
```

Comment:
should it be any import not only ReactPlayer ?

if no truncate split, should we consider it as an error and stop/fail the build (as what will happen ?)

Reply:
@benoitf I'll change it to exclude any import. Regarding the stop/fail build without  truncate, I think that could be a good idea since without this information, there will be no way to present the release notes banner. However, I'm not sure how significant is release notes banner because we might decide that we HAVE to show it with each release, so there is no option of building without the correct info/template, but if we decide that it'll be ok not showing them if there is an error, we can continue with the build as intended and add checks in the renderer to see if the info is available, and if not, just not show the banner.

Reply:
but how do you will consider that the data is "small enough" to be displayed for a release ?

like shouldn't it break the build if in the release notes, `<truncate>` is for example after too many lines or missing ?

else it would mean that client will have to display 1000 lines or the whole release notes.

Reply:
@benoitf since we only display the title and bullet points from the notes, I can limit how many lines to add to the json file for L48-49, for example take/use the first n lines from `summary` at L48.